File: api/src/main/java/com/neo4j/data/importer/Neo4jPredicate.java
package com.neo4j.data.importer;

import java.util.function.Predicate;

public class Neo4jPredicate {

    private final Predicate<Neo4j> predicate;

    Neo4jPredicate(Predicate<Neo4j> predicate) {
        this.predicate = predicate;
    }

    public boolean withNeo4j(Neo4j neo4j) {
        return predicate.test(neo4j);
    }

    public Neo4jPredicate and(Neo4jPredicate other) {
        return new Neo4jPredicate(predicate.and(other.predicate));
    }
}

File: api/src/main/kotlin/com/neo4j/data/importer/Neo4jPredicate.kt
package com.neo4j.data.importer

import java.util.function.Predicate

class Neo4jPredicate(private val predicate: Predicate<Neo4j>) {

  fun withNeo4j(neo4j: Neo4j): Boolean {
    return predicate.test(neo4j)
  }

  fun and(other: Neo4jPredicate): Neo4jPredicate {
    return Neo4jPredicate(predicate.and(other.predicate))
  }
}
